server-phoenix/lib/helios_web/controllers/web_hooks/slack_controller.ex

```diff
@@ -61,14 +61,20 @@ defmodule HeliosWeb.WebHooks.SlackController do
 
       insert_channel_id(channel_id)
 
+      event_channel_name =
+        SlackChannelNames
+        |> SlackChannelNames.name_to_id(channel_id)
+        |> Repo.one()
+
       unless Event
              |> Event.slack_messages()
              |> Event.with_external_id(event["event_ts"])
              |> Repo.exists?() do
         Repo.insert!(%Event{
           source: :slack_message,
           external_id: event["event_ts"],
-          source_author: event["user"]
+          source_author: event["user"],
+          source_channel: event_channel_name
```

Comment:
Names are usually a display level concern. My concern here is that the name of the channel can change, and then these events would all get orphaned. 

Would it make sense to store the `id` instead? Then when somebody asks for the list of events, we can join with the SlackChannelNames table to get the name from the id?

Reply:
hmmm... this event table seems more like a logger.  Perhaps, in this case, it's ok to have orphaned names when users &/or channels are removed or changed.